app/src/main/java/com/kms/cura/view/activity/MessageThreadActivity.java

```diff
@@ -131,13 +136,23 @@ public boolean onMenuItemClick(MenuItem item) {
     @Override
     public void onClick(View v) {
         if (v.getId() == R.id.btnReply) {
-            Toast.makeText(this, "Reply", Toast.LENGTH_SHORT).show();
-            // navigate to New Message
+            Intent intent = new Intent(this, NewMessageActivity.class);
+            if (CurrentUserProfile.getInstance().isPatient()) {
+                intent.putExtra(NewMessageActivity.KEY_SENDER, NewMessageActivity.KEY_PATIENT);
+                if (messageEntities.get(0).isSenderDoctor()) {
+                    intent.putExtra(NewMessageActivity.KEY_RECEIVER_ID, messageEntities.get(0).getSender().getId());
+                    intent.putExtra(NewMessageActivity.KEY_RECEIVER_NAME, messageEntities.get(0).getSender().getName());
+                } else {
+                    intent.putExtra(NewMessageActivity.KEY_RECEIVER_ID, messageEntities.get(0).getReceiver().getId());
+                    intent.putExtra(NewMessageActivity.KEY_RECEIVER_NAME, messageEntities.get(0).getReceiver().getName());
+                }
```

Comment:
instead of calling putExtra on if else like this, should try to set local variables of receiver_id and receiver_name base on the condition and then calling putExtra with those variables